api/src/pcapi/core/bookings/api.py

```diff
@@ -194,28 +197,9 @@ def _cancel_bookings_from_stock(stock: Stock, reason: BookingCancellationReasons
     Note that this will not reindex the stock.offer in Algolia
     """
     deleted_bookings: list[Booking] = []
-    with transaction():
-        stock = offers_repository.get_and_lock_stock(stock_id=stock.id)
-        for booking in stock.bookings:
-            try:
-                booking.cancel_booking()
-            except (BookingIsAlreadyUsed, BookingIsAlreadyCancelled) as e:
-                logger.info(str(e), extra={"booking": booking.id, "reason": str(reason)})
-            else:
-                booking.cancellationReason = reason
-                stock.dnBookedQuantity -= booking.quantity
-                deleted_bookings.append(booking)
-        repository.save(*deleted_bookings)
-
-    pro_emails = set()
-
-    for booking in deleted_bookings:
-        if booking.individualBooking is not None:
-            update_external_user(booking.individualBooking.user)
-            pro_emails.add(booking.venue.bookingEmail)
-
-    for pro_email in pro_emails:
-        update_external_pro(pro_email)
+    for booking in stock.bookings:
+        if _cancel_booking(booking, reason, stock.offer.isEvent):
```

Comment:
Utiliser un argument nommé me semblerait plus clair ici (pour qui ne connaît pas par coeur la signature de `_cancel_booking()` ! ;) ) : 

```suggestion
        if _cancel_booking(booking, reason, cancel_even_if_used=stock.offer.isEvent):
```